inst/include/sampler.h

```diff
@@ -71,7 +71,7 @@ namespace smc {
         ///Structure used internally for resampling.
         arma::vec dRSWeights;
         ///Structure used internally for resampling.
-        arma::Col<unsigned int> uRSCount;
+        arma::Col<int> uRSCount;
```

Comment:
A trivial detail, but the u prefix denotes an unsigned value...

Now this is based on some pretty old code and many don't like Hungarian notation any more. Which is fine, it's not something I'm religious about, but if we use it or something that looks like it then it should be meaningful. nRSCount would better match an object of this sort, but as soon as we're using arma:Col rather than a simple array even that's a bit of a stretch.

With that in mind, it would be good to think about how best to refer to name this one object to avoid introducing any unnecessary confusion and at some later stage (not necessarily within GSOC and this isn't your problem so much as an RcppSMC one) to think a bit about variable naming conventions.